### PR TITLE
simplified CMakeLists.txt

### PR DESCRIPTION
**Pull request by gf712**
@vigsterkr added cmake functions to remove some repetition

### CODE REVIEW

File: src/shogun/mathematics/graph/CMakeLists.txt [hunk 1]
```diff
@@ -71,70 +71,73 @@ set_target_properties(
...
+
+function(add_executor EXECUTOR_NAME)
+
+    cmake_parse_arguments(
+        EXECUTOR
+        ""
+        "ADDITIONAL_SHARED_LIBRARIES;ADDITIONAL_INCLUDE_DIRS"
+        "IMPLEMENTATION_HEADERS;IMPLEMENTATION_SOURCES"
+        ${ARGN}
+    )
```
Comment (gf712) [inline src/shogun/mathematics/graph/CMakeLists.txt:82]: @vigsterkr no idea if this how cmake_parse_arguments is meant to be used, but it seems to work

File: src/shogun/mathematics/graph/CMakeLists.txt [hunk 2]
```diff
@@ -71,70 +71,73 @@ set_target_properties(
...
-set_property(TARGET shogun-vanilla-executor PROPERTY POSITION_INDEPENDENT_CODE TRUE)
-set_target_properties(
-  shogun-vanilla-executor PROPERTIES
-  VERSION ${LIBSHOGUNVER}
-  SOVERSION ${LIBSHOGUNSO}
-  INTERFACE_shogun_MAJOR_VERSION ${SHOGUN_VERSION_MAJOR}
-  LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/src/shogun)
-set_property(TARGET shogun-vanilla-executor APPEND PROPERTY COMPATIBLE_INTERFACE_STRING ${SHOGUN_VERSION_MAJOR})
-LIST(APPEND EXCUTORS shogun-vanilla-executor)
+set(${EXECUTORS} "")
```
Comment (vigsterkr) [inline src/shogun/mathematics/graph/CMakeLists.txt]: this could be a LIST, to which you append: `LIST(APPEND ....`

**Review by vigsterkr — CHANGES_REQUESTED** [review]

File: src/shogun/mathematics/graph/CMakeLists.txt (hunk 2 above)
Comment (gf712) [reply src/shogun/mathematics/graph/CMakeLists.txt]: this was the only I found to add things to a variable in the parent scope https://cmake.org/pipermail/cmake/2018-September/068291.html

Comment (gf712) [reply src/shogun/mathematics/graph/CMakeLists.txt]: i removed the `set(${EXECUTORS} "")` as it is redundant.. In the function there is `set(EXECUTORS ${EXECUTORS} ${EXECUTOR_NAME} PARENT_SCOPE)`

**Pull request merged**